requirements-dev.txt

```diff
@@ -6,4 +6,7 @@ pytest-xdist>=2.5.0
 flaky>=3.7.0
 pytest-forked>=1.4.0
 black>=21
-requests~=2.28.1
\ No newline at end of file
+requests~=2.28.1
+pylint==2.7.4
+pre-commit==2.21.0
+dask==2022.12.1
```

Comment:
i'm not sure about these two. I think `pre-commit` is supposed to be an optional feature that you can install as suggested by the dev docs. I see `dask` is already in the other 2 requirements.txt files but with the `[delayed]` option, so at a minimum that should be here too.

Generally, I'm not sure what the difference is between this file and `requirements.txt` though. What's that one used for if not development? The installation/use of `pennylane` itself shouldn't depend on it.

Reply:
Seems like all the requirement files are out of date. We have a ticket to fix it along with the Developer guide. I will add the [delayed] flag though

Reply:
Going to remove pre-commit and dask, pre-commit is already specified at the top of the file, and dask can be pulled from requirement.txt. We'll consolidate the two requirements files separately.